service/history/workflow/context.go

```diff
@@ -669,14 +670,14 @@ func (c *ContextImpl) UpdateWorkflowExecutionWithNew(
 	return nil
 }
 
-func (c *ContextImpl) SetWorkflowExecution(ctx context.Context, now time.Time) (retError error) {
+func (c *ContextImpl) SetWorkflowExecution(ctx context.Context, now time.Time, mutableState MutableState) (retError error) {
```

Comment:
I think I am not following? If the goal is to the mutable state to be something else. We should clear the cache when the call succeeds?

Reply:
Sync offline. This change is not needed